pkg/store/store.go

```diff
@@ -44,21 +44,33 @@ func (s *DefaultStore) Close() error {
 }
 
 // SetHeight sets the height saved in the Store if it is higher than the existing height
-func (s *DefaultStore) SetHeight(ctx context.Context, height uint64) {
-	for {
-		storeHeight := s.height.Load()
-		if height <= storeHeight {
-			break
-		}
-		if s.height.CompareAndSwap(storeHeight, height) {
-			break
-		}
+func (s *DefaultStore) SetHeight(ctx context.Context, height uint64) error {
+	currentHeight := s.Height(ctx)
+	if height <= currentHeight {
+		return nil
 	}
+
+	heightBytes := encodeHeight(height)
+	return s.db.Put(ctx, ds.NewKey(getHeightKey()), heightBytes)
 }
 
 // Height returns height of the highest block saved in the Store.
-func (s *DefaultStore) Height() uint64 {
-	return s.height.Load()
+func (s *DefaultStore) Height(ctx context.Context) uint64 {
+	heightBytes, err := s.db.Get(ctx, ds.NewKey(getHeightKey()))
+	if errors.Is(err, ds.ErrNotFound) {
```

Comment:
why handpick `ErrNotFound` if the intention is to return `0` as default?

Reply:
just leaving a comment explaining the rationale for returning 0 should be good enough.

Reply:
I think we need to return an error here, will refactor

Reply:
Ah ok, if err is not found we return 0 without errors, because it could be the starting of a chain. Then for the other case we should return an error

Reply:
if it is inconvenient to return error, you could log a warning for the later case and return 0. it should be ok.